### PR TITLE
[RayJob] add Failing RayJob in HTTPMode e2e test for rayjob with retry

### PR DESCRIPTION
**Pull request by tinaxfwu**
## Why are these changes needed?

Add the "The RayJob in HTTPMode fails and retries" test.

https://github.com/ray-project/kuberay/blob/ea314d7ee780e840382f3de852918f69d7867f25/ray-operator/test/e2e/rayjob_retry_test.go#L89-L93

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

Testing was conducted on my Apply M2 Mac, with the RayImage changed to `rayproject/ray:2.9.0-aarch64` in the `ray-operator/test/support/defaults.go` file.

Below are the commands I used for the test and the resulting test logs. 
```
$ cd ray-operator
$ kind create cluster --image=kindest/node:v1.24.0
$ IMG=kuberay/operator:nightly make docker-build
$ kind load docker-image kuberay/operator:nightly
$ helm install kuberay-operator --set image.repository=kuberay/operator --set image.tag=nightly ../helm-chart/kuberay-operator
$ go test -timeout 30m -v ./test/e2e/rayjob_retry_test.go ./test/e2e/support.go 
```
```
--- PASS: TestRayJobRetry (283.95s)
    --- PASS: TestRayJobRetry/Failing_RayJob_without_cluster_shutdown_after_finished (134.35s)
    --- PASS: TestRayJobRetry/Failing_submitter_K8s_Job (76.77s)
    --- PASS: TestRayJobRetry/Failing_RayJob_in_HTTPMode (72.70s)
PASS
ok  	command-line-arguments	284.457s
```

### CODE REVIEW

File: ray-operator/test/e2e/rayjob_retry_test.go [hunk 1]
```diff
@@ -133,9 +133,45 @@ func TestRayJobRetry(t *testing.T) {
 	// regardless of the value of backoffLimit. Refer to rayjob_test.go for an example.
 	// })
 
-	// test.T().Run("Failing RayJob in HTTPMode", func(_ *testing.T) {
-	// TODO: The RayJob in HTTPMode fails and retries. This test is similar to the
-	// "Failing RayJob without cluster shutdown after finished" test in rayjob_lightweight_test.go.
-	// The difference is that here we set RayJob.BackoffLimit.
-	// })
+	test.T().Run("Failing RayJob in HTTPMode", func(_ *testing.T) {
```
Comment (andrewsykim) [inline ray-operator/test/e2e/rayjob_retry_test.go:136]: nit: Failing RayJob with HttpMode submission mode

**Review by andrewsykim — APPROVED** [review]
LGTM, thanks!

File: ray-operator/test/e2e/rayjob_retry_test.go [hunk 2]
```diff
@@ -133,9 +133,45 @@ func TestRayJobRetry(t *testing.T) {
...
+				WithEntrypoint("python /home/ray/jobs/fail.py").
+				WithShutdownAfterJobFinishes(false).
+				WithRayClusterSpec(newRayClusterSpec(mountConfigMap[rayv1ac.RayClusterSpecApplyConfiguration](jobs, "/home/ray/jobs"))))
+
+		rayJob, err := test.Client().Ray().RayV1().RayJobs(namespace.Name).Apply(test.Ctx(), rayJobAC, TestApplyOptions)
+		test.Expect(err).NotTo(HaveOccurred())
+		test.T().Logf("Created RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
+
+		test.T().Logf("Waiting for RayJob %s/%s to complete", rayJob.Namespace, rayJob.Name)
+		test.Eventually(RayJob(test, rayJob.Namespace, rayJob.Name), TestTimeoutMedium).
```
Comment (kevin85421) [inline ray-operator/test/e2e/rayjob_retry_test.go:151]: This seems to be unnecessary in this case.

**Comment by kevin85421** [comment]
@tinaxfwu can you fix the lint error? It's helpful to install pre-commit https://github.com/ray-project/kuberay/blob/master/ray-operator/DEVELOPMENT.md#pre-commit-hooks.

**Comment by tinaxfwu** [comment]
@kevin85421 I fixed the lint error, but the` buildkite/ray-ecosystem-ci-kuberay-ci/test-raycluster-sample-yamls-latest-release` check. I'm not quite sure I understand this issue or how to fix it.

**Review by kevin85421 — APPROVED** [review]

**Comment by kevin85421** [comment]
@tinaxfwu the failure is not relevant. It is just a flaky test.

**Pull request merged**